.github/workflows/build_and_test.yml

```diff
@@ -93,41 +83,62 @@ jobs:
     - name: Check labeler configuration
       run: scripts/check-labeler.pl .github/labeler.yml
 
-    - name: Unit tests
-      env:
-        KONG_TEST_PG_DATABASE: kong
-        KONG_TEST_PG_USER: kong
+  schedule:
+    name: Schedule busted tests to run
+    runs-on: ubuntu-22.04
+    needs: build
+
+    env:
+      WORKFLOW_ID: ${{ github.run_id }}
+
+    outputs:
+      runners: ${{ steps.generate-runner-array.outputs.RUNNERS }}
+
+    steps:
+    - name: Checkout source code
+      uses: actions/checkout@v4
+
+    - name: Set up NodeJS
+      uses: actions/setup-node@v4
+      with:
+        node-version: 21
+        cache: 'npm'
+        cache-dependency-path: .ci/schedule-tests/package-lock.json
+
+    - name: Schedule tests
       run: |
-          source ${{ env.BUILD_ROOT }}/kong-dev-venv.sh
-          TEST_CMD="bin/busted -v -o htest spec/01-unit"
-          if [[ $KONG_TEST_COVERAGE = true ]]; then
-            TEST_CMD="$TEST_CMD --coverage"
-          fi
-          $TEST_CMD
+        cd .ci/schedule-tests/
+        npm install
+        node src/main.js schedule ../test_suites.json ../runtimes.txt ../../test-chunk. $RUNNER_COUNT
```

Comment:
it would be awesome if this invocation (and the later when actually scheudling the test to run) can be made into a actions instead of relying on users to call directly. there are couple of benefits: 
1) later on you don't need to explictly do things like `actions/setup-node` from caller side; invocating other actions like upload-artifacts can also be self-contained.
2) from time to time we can make sure we are not leaking or misuse of an env var from the parent workflow.
3) it will be easy to make a switch to turn on/off the scheduling if anything goes off in urgent cases

Reply:
The Javascript parts have moved to a new repository: https://github.com/Kong/gateway-test-scheduler